### PR TITLE
CMake: Include instead of copying cpu kernel files

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #67656
* #67655
* #68555

Currently, each cpu kernel file is copied into the build folder 3 times to give them different compilation flags. This changes it to instead generate 3 files that `#include` the original file. The biggest difference is that updating a copied file requires `cmake` to re-run, whereas include dependencies are natively handled by `ninja`.

A side benefit is that included files show up directly in the build dependency graph, whereas `cmake` file copies don't.

Differential Revision: [D32566108](https://our.internmc.facebook.com/intern/diff/D32566108)

### CODE REVIEW

**Review by malfet — APPROVED** [review]
LGTM

File: cmake/IncludeSource.cpp.in [hunk 1]
```diff
@@ -0,0 +1 @@
+#include "@IMPL@"
```
Comment (malfet) [inline cmake/IncludeSource.cpp.in:1]: Nit (This way we would not need to build separate compilation rules for every CPU capability)
```suggestion
#define CPU_CAPABILITY @CPU_CAPABILITY@
#include "@IMPL@"
```

Comment (peterbell10) [reply cmake/IncludeSource.cpp.in:1]: I had considered that, but we still need to pass the vectorization flags like `-march=avx2` so I think It makes sense to define everything `CPU_CAPABILITY` related on the command line.

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32566108).

**Pull request closed without merging**